### PR TITLE
Add new helm chart tidb-drainer to support multiple drainer

### LINKED ISSUE #661: Support multiple drainers
## Feature Request

**Is your feature request related to a problem? Please describe:**
Currently you can only deploy one drainer per TiDB cluster.

**Describe the feature you'd like:**
Ability to deploy multiple drainers per TiDB cluster.

**Describe alternatives you've considered:**
Chaining multiple TiDB clusters would work, but it's not ideal - increased dependency/complexity and potential lag problem.

**Teachability, Documentation, Adoption, Migration Strategy:**
Example scenario would be:
1. Use one drainer to replicate to another TiDB cluster or MySQL; and
2. Use another drainer to dump incremental backup

Just like how the architecture diagram shows here:
https://pingcap.com/docs/v3.0/reference/tidb-binlog-overview/?origin_team=T0351U5DY#tidb-binlog-cluster-overview

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

### What problem does this PR solve? 
close #661 

### What is changed and how does it work?
Add a new helm chart `tidb-drainer`.

The documentation PR: https://github.com/pingcap/docs/pull/1440

The stability test and e2e-test will be added in a follow up PR.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Helm charts change

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
Introduce a new helm chart,  tidb-drainer, to facilitate the multiple drainers management.
 ```

@weekface @tennix @xiaojingchen PTAL

### CODE REVIEW

File: charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+done
+
+/drainer \
+-L={{ .Values.logLevel | default "info" }} \
+-pd-urls=http://{{ .Values.clusterName }}-pd:2379 \
+-addr=`echo ${HOSTNAME}`.{{ include "drainer.name" . }}:8249 \
+-config=/etc/drainer/drainer.toml \
+-disable-detect={{ .Values.disableDetect | default false }} \
+-initial-commit-ts={{ .Values.initialCommitTs | default 0 }} \
+-log-file=
```
Comment (gregwebs) [inline charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl]: This empty argument looks odd.

File: charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl [hunk 2]
```diff
@@ -0,0 +1,34 @@
+set -euo pipefail
+
+domain=`echo ${HOSTNAME}`.{{ include "drainer.name" . }}
+
+elapseTime=0
+period=1
+threshold=30
+while true; do
```
Comment (gregwebs) [inline charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl]: This part could be an init container. Also it looks like this technically would need the wait-for-pd init container from https://github.com/pingcap/tidb-operator/pull/581 if someone wanted to install this helm chart immediately after the tidb-cluster chart.

File: charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl (hunk 1 above)
Comment (aylei) [reply charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl]: Yes, pass an empty string explicitly would be better

File: charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl (hunk 2 above)
Comment (aylei) [reply charts/tidb-drainer/templates/scripts/_start_drainer.sh.tpl]: Agreed, tracked in #749 to keep this PR simple (actually this PR just extracts the drainer statefulset from the tidb-cluster chart)

**Comment by weekface** [comment]
You should create two issues to track things:
- add e2e tests
- add chart release
- other?

File: charts/tidb-drainer/templates/drainer-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: {{ include "drainer-configmap.name" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "drainer-configmap.name" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: drainer
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+data:
+{{ include "drainer-configmap.data" . | indent 2 }}
```
Comment (weekface) [inline charts/tidb-drainer/templates/drainer-configmap.yaml]: add new line

File: charts/tidb-drainer/templates/drainer-service.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  clusterIP: None
+  ports:
+  - name: drainer
+    port: 8249
+  selector:
+    app.kubernetes.io/name: {{ include "drainer.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: drainer
```
Comment (weekface) [inline charts/tidb-drainer/templates/drainer-service.yaml]: new line

File: charts/tidb-drainer/templates/drainer-statefulset.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: drainer
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "drainer.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/managed-by: tidb-operator
```
Comment (weekface) [inline charts/tidb-drainer/templates/drainer-statefulset.yaml]: managed-by: tidb-operator?

File: charts/tidb-drainer/templates/drainer-statefulset.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path: "/metrics"
+        prometheus.io/port: "8249"
+      labels:
+        app.kubernetes.io/name: {{ include "drainer.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/managed-by: tidb-operator
```
Comment (weekface) [inline charts/tidb-drainer/templates/drainer-statefulset.yaml]: ditto

File: charts/tidb-drainer/values.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  [syncer]
+  worker-count = 16
+  detect-interval = 10
+  disable-dispatch = false
+  ignore-schemas = "INFORMATION_SCHEMA,PERFORMANCE_SCHEMA,mysql"
+  safe-mode = false
+  txn-batch = 20
+  db-type = "pb"
+  [syncer.to]
+  dir = "/data/pb"
```
Comment (weekface) [inline charts/tidb-drainer/values.yaml]: use mysql instead?: https://github.com/pingcap/tidb-operator/pull/693/files#diff-b8571ebe161ca414e35a16dc83482915R450-R455

File: charts/tidb-drainer/values.yaml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #  cpu: 100m
+  #  memory: 128Mi
+  # requests:
+  #  cpu: 100m
+nodeSelector: {}
+
+tolerations: []
```
Comment (weekface) [inline charts/tidb-drainer/values.yaml]: `tolerations ` and `affinity` is used?

File: charts/tidb-drainer/values.yaml (hunk 7 above)
Comment (aylei) [reply charts/tidb-drainer/values.yaml]: `mysql` cannot work out of box unless the user configure the mysql connection properly

**Comment by aylei** [comment]
@weekface All comments are addressed, PTAL again

**Comment by aylei** [comment]
@tennix @weekface I've fixed a fmt error which blocks the CI, PTAL again

**Pull request merged**